// coreum-js@2.17.4/dist/main/coreum/customparams/v1/genesis.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.0
//   protoc               v3.21.12
// source: coreum-protos/customparams/genesis.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenesisState = exports.protobufPackage = void 0;
/* eslint-disable */
const binary_1 = require("cosmjs-types/binary");
const params_1 = require("./params");
exports.protobufPackage = "coreum.customparams.v1";
function createBaseGenesisState() {
    return { stakingParams: undefined };
}
exports.GenesisState = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.stakingParams !== undefined) {
            params_1.StakingParams.encode(message.stakingParams, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGenesisState();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.stakingParams = params_1.StakingParams.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            stakingParams: isSet(object.stakingParams)
                ? params_1.StakingParams.fromJSON(object.stakingParams)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.stakingParams !== undefined) {
            obj.stakingParams = params_1.StakingParams.toJSON(message.stakingParams);
        }
        return obj;
    },
    create(base) {
        return exports.GenesisState.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGenesisState();
        message.stakingParams =
            object.stakingParams !== undefined && object.stakingParams !== null
                ? params_1.StakingParams.fromPartial(object.stakingParams)
                : undefined;
        return message;
    },
};
function isSet(value) {
    return value !== null && value !== undefined;
}
