// coreum-js@2.17.4/dist/module/coreum/asset/ft/v1/params.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.0
//   protoc               v3.21.12
// source: coreum-protos/ft/params.proto
/* eslint-disable */
import { BinaryReader, BinaryWriter } from "cosmjs-types/binary";
import { Coin } from "cosmjs-types/cosmos/base/v1beta1/coin";
import { Duration } from "cosmjs-types/google/protobuf/duration";
import { Timestamp } from "cosmjs-types/google/protobuf/timestamp";
export const protobufPackage = "coreum.asset.ft.v1";
function createBaseParams() {
    return {
        issueFee: undefined,
        tokenUpgradeDecisionTimeout: undefined,
        tokenUpgradeGracePeriod: undefined,
    };
}
export const Params = {
    encode(message, writer = new BinaryWriter()) {
        if (message.issueFee !== undefined) {
            Coin.encode(message.issueFee, writer.uint32(10).fork()).ldelim();
        }
        if (message.tokenUpgradeDecisionTimeout !== undefined) {
            Timestamp.encode(toTimestamp(message.tokenUpgradeDecisionTimeout), writer.uint32(18).fork()).ldelim();
        }
        if (message.tokenUpgradeGracePeriod !== undefined) {
            Duration.encode(message.tokenUpgradeGracePeriod, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof BinaryReader ? input : new BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.issueFee = Coin.decode(reader, reader.uint32());
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.tokenUpgradeDecisionTimeout = fromTimestamp(Timestamp.decode(reader, reader.uint32()));
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.tokenUpgradeGracePeriod = Duration.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            issueFee: isSet(object.issueFee)
                ? Coin.fromJSON(object.issueFee)
                : undefined,
            tokenUpgradeDecisionTimeout: isSet(object.tokenUpgradeDecisionTimeout)
                ? fromJsonTimestamp(object.tokenUpgradeDecisionTimeout)
                : undefined,
            tokenUpgradeGracePeriod: isSet(object.tokenUpgradeGracePeriod)
                ? Duration.fromJSON(object.tokenUpgradeGracePeriod)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.issueFee !== undefined) {
            obj.issueFee = Coin.toJSON(message.issueFee);
        }
        if (message.tokenUpgradeDecisionTimeout !== undefined) {
            obj.tokenUpgradeDecisionTimeout =
                message.tokenUpgradeDecisionTimeout.toISOString();
        }
        if (message.tokenUpgradeGracePeriod !== undefined) {
            obj.tokenUpgradeGracePeriod = Duration.toJSON(message.tokenUpgradeGracePeriod);
        }
        return obj;
    },
    create(base) {
        return Params.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseParams();
        message.issueFee =
            object.issueFee !== undefined && object.issueFee !== null
                ? Coin.fromPartial(object.issueFee)
                : undefined;
        message.tokenUpgradeDecisionTimeout =
            object.tokenUpgradeDecisionTimeout ?? undefined;
        message.tokenUpgradeGracePeriod =
            object.tokenUpgradeGracePeriod !== undefined &&
                object.tokenUpgradeGracePeriod !== null
                ? Duration.fromPartial(object.tokenUpgradeGracePeriod)
                : undefined;
        return message;
    },
};
function toTimestamp(date) {
    const seconds = Math.trunc(date.getTime() / 1_000);
    const nanos = (date.getTime() % 1_000) * 1_000_000;
    return { seconds: BigInt(seconds), nanos };
}
function fromTimestamp(t) {
    const seconds = BigInt(t.seconds || 0);
    const nanos = BigInt(t.nanos || 0);
    const millis = seconds * BigInt(1_000) + nanos / BigInt(1_000_000);
    return new Date(Number(millis)); // Convert bigint to number
}
function fromJsonTimestamp(o) {
    if (o instanceof globalThis.Date) {
        return o;
    }
    else if (typeof o === "string") {
        return new globalThis.Date(o);
    }
    else {
        return fromTimestamp(Timestamp.fromJSON(o));
    }
}
function isSet(value) {
    return value !== null && value !== undefined;
}
