// coreum-js@2.17.4/dist/main/coreum/feemodel/v1/query.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.0
//   protoc               v3.21.12
// source: coreum-protos/feemodel/query.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryClientImpl = exports.QueryServiceName = exports.QueryParamsResponse = exports.QueryParamsRequest = exports.QueryRecommendedGasPriceResponse = exports.QueryRecommendedGasPriceRequest = exports.QueryMinGasPriceResponse = exports.QueryMinGasPriceRequest = exports.protobufPackage = void 0;
/* eslint-disable */
const binary_1 = require("cosmjs-types/binary");
const coin_1 = require("cosmjs-types/cosmos/base/v1beta1/coin");
const params_1 = require("./params");
exports.protobufPackage = "coreum.feemodel.v1";
function createBaseQueryMinGasPriceRequest() {
    return {};
}
exports.QueryMinGasPriceRequest = {
    encode(_, writer = new binary_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryMinGasPriceRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.QueryMinGasPriceRequest.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseQueryMinGasPriceRequest();
        return message;
    },
};
function createBaseQueryMinGasPriceResponse() {
    return { minGasPrice: undefined };
}
exports.QueryMinGasPriceResponse = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.minGasPrice !== undefined) {
            coin_1.DecCoin.encode(message.minGasPrice, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryMinGasPriceResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.minGasPrice = coin_1.DecCoin.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            minGasPrice: isSet(object.minGasPrice)
                ? coin_1.DecCoin.fromJSON(object.minGasPrice)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.minGasPrice !== undefined) {
            obj.minGasPrice = coin_1.DecCoin.toJSON(message.minGasPrice);
        }
        return obj;
    },
    create(base) {
        return exports.QueryMinGasPriceResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryMinGasPriceResponse();
        message.minGasPrice =
            object.minGasPrice !== undefined && object.minGasPrice !== null
                ? coin_1.DecCoin.fromPartial(object.minGasPrice)
                : undefined;
        return message;
    },
};
function createBaseQueryRecommendedGasPriceRequest() {
    return { afterBlocks: 0 };
}
exports.QueryRecommendedGasPriceRequest = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.afterBlocks !== 0) {
            writer.uint32(8).uint32(message.afterBlocks);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryRecommendedGasPriceRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 8) {
                        break;
                    }
                    message.afterBlocks = reader.uint32();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            afterBlocks: isSet(object.afterBlocks)
                ? globalThis.Number(object.afterBlocks)
                : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.afterBlocks !== 0) {
            obj.afterBlocks = Math.round(message.afterBlocks);
        }
        return obj;
    },
    create(base) {
        return exports.QueryRecommendedGasPriceRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryRecommendedGasPriceRequest();
        message.afterBlocks = object.afterBlocks ?? 0;
        return message;
    },
};
function createBaseQueryRecommendedGasPriceResponse() {
    return { low: undefined, med: undefined, high: undefined };
}
exports.QueryRecommendedGasPriceResponse = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.low !== undefined) {
            coin_1.DecCoin.encode(message.low, writer.uint32(10).fork()).ldelim();
        }
        if (message.med !== undefined) {
            coin_1.DecCoin.encode(message.med, writer.uint32(18).fork()).ldelim();
        }
        if (message.high !== undefined) {
            coin_1.DecCoin.encode(message.high, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryRecommendedGasPriceResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.low = coin_1.DecCoin.decode(reader, reader.uint32());
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.med = coin_1.DecCoin.decode(reader, reader.uint32());
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.high = coin_1.DecCoin.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            low: isSet(object.low) ? coin_1.DecCoin.fromJSON(object.low) : undefined,
            med: isSet(object.med) ? coin_1.DecCoin.fromJSON(object.med) : undefined,
            high: isSet(object.high) ? coin_1.DecCoin.fromJSON(object.high) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.low !== undefined) {
            obj.low = coin_1.DecCoin.toJSON(message.low);
        }
        if (message.med !== undefined) {
            obj.med = coin_1.DecCoin.toJSON(message.med);
        }
        if (message.high !== undefined) {
            obj.high = coin_1.DecCoin.toJSON(message.high);
        }
        return obj;
    },
    create(base) {
        return exports.QueryRecommendedGasPriceResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryRecommendedGasPriceResponse();
        message.low =
            object.low !== undefined && object.low !== null
                ? coin_1.DecCoin.fromPartial(object.low)
                : undefined;
        message.med =
            object.med !== undefined && object.med !== null
                ? coin_1.DecCoin.fromPartial(object.med)
                : undefined;
        message.high =
            object.high !== undefined && object.high !== null
                ? coin_1.DecCoin.fromPartial(object.high)
                : undefined;
        return message;
    },
};
function createBaseQueryParamsRequest() {
    return {};
}
exports.QueryParamsRequest = {
    encode(_, writer = new binary_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryParamsRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.QueryParamsRequest.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseQueryParamsRequest();
        return message;
    },
};
function createBaseQueryParamsResponse() {
    return { params: undefined };
}
exports.QueryParamsResponse = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.params !== undefined) {
            params_1.Params.encode(message.params, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryParamsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.params = params_1.Params.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            params: isSet(object.params) ? params_1.Params.fromJSON(object.params) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.params !== undefined) {
            obj.params = params_1.Params.toJSON(message.params);
        }
        return obj;
    },
    create(base) {
        return exports.QueryParamsResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryParamsResponse();
        message.params =
            object.params !== undefined && object.params !== null
                ? params_1.Params.fromPartial(object.params)
                : undefined;
        return message;
    },
};
exports.QueryServiceName = "coreum.feemodel.v1.Query";
class QueryClientImpl {
    constructor(rpc, opts) {
        this.service = opts?.service || exports.QueryServiceName;
        this.rpc = rpc;
        this.MinGasPrice = this.MinGasPrice.bind(this);
        this.RecommendedGasPrice = this.RecommendedGasPrice.bind(this);
        this.Params = this.Params.bind(this);
    }
    MinGasPrice(request) {
        const data = exports.QueryMinGasPriceRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "MinGasPrice", data);
        return promise.then((data) => exports.QueryMinGasPriceResponse.decode(new binary_1.BinaryReader(data)));
    }
    RecommendedGasPrice(request) {
        const data = exports.QueryRecommendedGasPriceRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "RecommendedGasPrice", data);
        return promise.then((data) => exports.QueryRecommendedGasPriceResponse.decode(new binary_1.BinaryReader(data)));
    }
    Params(request) {
        const data = exports.QueryParamsRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "Params", data);
        return promise.then((data) => exports.QueryParamsResponse.decode(new binary_1.BinaryReader(data)));
    }
}
exports.QueryClientImpl = QueryClientImpl;
function isSet(value) {
    return value !== null && value !== undefined;
}
