// coreum-js@2.17.4/dist/module/coreum/asset/nft/v1/params.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.0
//   protoc               v3.21.12
// source: coreum-protos/nft/params.proto
/* eslint-disable */
import { BinaryReader, BinaryWriter } from "cosmjs-types/binary";
import { Coin } from "cosmjs-types/cosmos/base/v1beta1/coin";
export const protobufPackage = "coreum.asset.nft.v1";
function createBaseParams() {
    return { mintFee: undefined };
}
export const Params = {
    encode(message, writer = new BinaryWriter()) {
        if (message.mintFee !== undefined) {
            Coin.encode(message.mintFee, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof BinaryReader ? input : new BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.mintFee = Coin.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            mintFee: isSet(object.mintFee)
                ? Coin.fromJSON(object.mintFee)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.mintFee !== undefined) {
            obj.mintFee = Coin.toJSON(message.mintFee);
        }
        return obj;
    },
    create(base) {
        return Params.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseParams();
        message.mintFee =
            object.mintFee !== undefined && object.mintFee !== null
                ? Coin.fromPartial(object.mintFee)
                : undefined;
        return message;
    },
};
function isSet(value) {
    return value !== null && value !== undefined;
}
