// coreum-js@2.17.4/dist/main/coreum/asset/ft/v1/authz.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.0
//   protoc               v3.21.12
// source: coreum-protos/ft/authz.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.BurnAuthorization = exports.MintAuthorization = exports.protobufPackage = void 0;
/* eslint-disable */
const binary_1 = require("cosmjs-types/binary");
const coin_1 = require("cosmjs-types/cosmos/base/v1beta1/coin");
exports.protobufPackage = "coreum.asset.ft.v1";
function createBaseMintAuthorization() {
    return { mintLimit: [] };
}
exports.MintAuthorization = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        for (const v of message.mintLimit) {
            coin_1.Coin.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMintAuthorization();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.mintLimit.push(coin_1.Coin.decode(reader, reader.uint32()));
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            mintLimit: globalThis.Array.isArray(object?.mintLimit)
                ? object.mintLimit.map((e) => coin_1.Coin.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.mintLimit?.length) {
            obj.mintLimit = message.mintLimit.map((e) => coin_1.Coin.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.MintAuthorization.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMintAuthorization();
        message.mintLimit = object.mintLimit?.map((e) => coin_1.Coin.fromPartial(e)) || [];
        return message;
    },
};
function createBaseBurnAuthorization() {
    return { burnLimit: [] };
}
exports.BurnAuthorization = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        for (const v of message.burnLimit) {
            coin_1.Coin.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseBurnAuthorization();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.burnLimit.push(coin_1.Coin.decode(reader, reader.uint32()));
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            burnLimit: globalThis.Array.isArray(object?.burnLimit)
                ? object.burnLimit.map((e) => coin_1.Coin.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.burnLimit?.length) {
            obj.burnLimit = message.burnLimit.map((e) => coin_1.Coin.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.BurnAuthorization.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseBurnAuthorization();
        message.burnLimit = object.burnLimit?.map((e) => coin_1.Coin.fromPartial(e)) || [];
        return message;
    },
};
