// coreum-js@2.17.4/dist/main/coreum/asset/ft/v1/tx.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.0
//   protoc               v3.21.12
// source: coreum-protos/ft/tx.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgClientImpl = exports.MsgServiceName = exports.EmptyResponse = exports.MsgUpdateDEXWhitelistedDenoms = exports.MsgUpdateDEXUnifiedRefAmount = exports.MsgUpdateParams = exports.MsgUpgradeTokenV1 = exports.MsgClearAdmin = exports.MsgTransferAdmin = exports.MsgSetWhitelistedLimit = exports.MsgClawback = exports.MsgGloballyUnfreeze = exports.MsgGloballyFreeze = exports.MsgSetFrozen = exports.MsgUnfreeze = exports.MsgFreeze = exports.MsgBurn = exports.MsgMint = exports.ExtensionIssueSettings = exports.MsgIssue = exports.protobufPackage = void 0;
/* eslint-disable */
const binary_1 = require("cosmjs-types/binary");
const coin_1 = require("cosmjs-types/cosmos/base/v1beta1/coin");
const params_1 = require("./params");
const token_1 = require("./token");
exports.protobufPackage = "coreum.asset.ft.v1";
function createBaseMsgIssue() {
    return {
        issuer: "",
        symbol: "",
        subunit: "",
        precision: 0,
        initialAmount: "",
        description: "",
        features: [],
        burnRate: "",
        sendCommissionRate: "",
        uri: "",
        uriHash: "",
        extensionSettings: undefined,
        dexSettings: undefined,
    };
}
exports.MsgIssue = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.issuer !== "") {
            writer.uint32(10).string(message.issuer);
        }
        if (message.symbol !== "") {
            writer.uint32(18).string(message.symbol);
        }
        if (message.subunit !== "") {
            writer.uint32(26).string(message.subunit);
        }
        if (message.precision !== 0) {
            writer.uint32(32).uint32(message.precision);
        }
        if (message.initialAmount !== "") {
            writer.uint32(42).string(message.initialAmount);
        }
        if (message.description !== "") {
            writer.uint32(50).string(message.description);
        }
        writer.uint32(58).fork();
        for (const v of message.features) {
            writer.int32(v);
        }
        writer.ldelim();
        if (message.burnRate !== "") {
            writer.uint32(66).string(message.burnRate);
        }
        if (message.sendCommissionRate !== "") {
            writer.uint32(74).string(message.sendCommissionRate);
        }
        if (message.uri !== "") {
            writer.uint32(82).string(message.uri);
        }
        if (message.uriHash !== "") {
            writer.uint32(90).string(message.uriHash);
        }
        if (message.extensionSettings !== undefined) {
            exports.ExtensionIssueSettings.encode(message.extensionSettings, writer.uint32(98).fork()).ldelim();
        }
        if (message.dexSettings !== undefined) {
            token_1.DEXSettings.encode(message.dexSettings, writer.uint32(106).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgIssue();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.issuer = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.symbol = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.subunit = reader.string();
                    continue;
                }
                case 4: {
                    if (tag !== 32) {
                        break;
                    }
                    message.precision = reader.uint32();
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.initialAmount = reader.string();
                    continue;
                }
                case 6: {
                    if (tag !== 50) {
                        break;
                    }
                    message.description = reader.string();
                    continue;
                }
                case 7: {
                    if (tag === 56) {
                        message.features.push(reader.int32());
                        continue;
                    }
                    if (tag === 58) {
                        const end2 = reader.uint32() + reader.pos;
                        while (reader.pos < end2) {
                            message.features.push(reader.int32());
                        }
                        continue;
                    }
                    break;
                }
                case 8: {
                    if (tag !== 66) {
                        break;
                    }
                    message.burnRate = reader.string();
                    continue;
                }
                case 9: {
                    if (tag !== 74) {
                        break;
                    }
                    message.sendCommissionRate = reader.string();
                    continue;
                }
                case 10: {
                    if (tag !== 82) {
                        break;
                    }
                    message.uri = reader.string();
                    continue;
                }
                case 11: {
                    if (tag !== 90) {
                        break;
                    }
                    message.uriHash = reader.string();
                    continue;
                }
                case 12: {
                    if (tag !== 98) {
                        break;
                    }
                    message.extensionSettings = exports.ExtensionIssueSettings.decode(reader, reader.uint32());
                    continue;
                }
                case 13: {
                    if (tag !== 106) {
                        break;
                    }
                    message.dexSettings = token_1.DEXSettings.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            issuer: isSet(object.issuer) ? globalThis.String(object.issuer) : "",
            symbol: isSet(object.symbol) ? globalThis.String(object.symbol) : "",
            subunit: isSet(object.subunit) ? globalThis.String(object.subunit) : "",
            precision: isSet(object.precision)
                ? globalThis.Number(object.precision)
                : 0,
            initialAmount: isSet(object.initialAmount)
                ? globalThis.String(object.initialAmount)
                : "",
            description: isSet(object.description)
                ? globalThis.String(object.description)
                : "",
            features: globalThis.Array.isArray(object?.features)
                ? object.features.map((e) => (0, token_1.featureFromJSON)(e))
                : [],
            burnRate: isSet(object.burnRate)
                ? globalThis.String(object.burnRate)
                : "",
            sendCommissionRate: isSet(object.sendCommissionRate)
                ? globalThis.String(object.sendCommissionRate)
                : "",
            uri: isSet(object.uri) ? globalThis.String(object.uri) : "",
            uriHash: isSet(object.uriHash) ? globalThis.String(object.uriHash) : "",
            extensionSettings: isSet(object.extensionSettings)
                ? exports.ExtensionIssueSettings.fromJSON(object.extensionSettings)
                : undefined,
            dexSettings: isSet(object.dexSettings)
                ? token_1.DEXSettings.fromJSON(object.dexSettings)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.issuer !== "") {
            obj.issuer = message.issuer;
        }
        if (message.symbol !== "") {
            obj.symbol = message.symbol;
        }
        if (message.subunit !== "") {
            obj.subunit = message.subunit;
        }
        if (message.precision !== 0) {
            obj.precision = Math.round(message.precision);
        }
        if (message.initialAmount !== "") {
            obj.initialAmount = message.initialAmount;
        }
        if (message.description !== "") {
            obj.description = message.description;
        }
        if (message.features?.length) {
            obj.features = message.features.map((e) => (0, token_1.featureToJSON)(e));
        }
        if (message.burnRate !== "") {
            obj.burnRate = message.burnRate;
        }
        if (message.sendCommissionRate !== "") {
            obj.sendCommissionRate = message.sendCommissionRate;
        }
        if (message.uri !== "") {
            obj.uri = message.uri;
        }
        if (message.uriHash !== "") {
            obj.uriHash = message.uriHash;
        }
        if (message.extensionSettings !== undefined) {
            obj.extensionSettings = exports.ExtensionIssueSettings.toJSON(message.extensionSettings);
        }
        if (message.dexSettings !== undefined) {
            obj.dexSettings = token_1.DEXSettings.toJSON(message.dexSettings);
        }
        return obj;
    },
    create(base) {
        return exports.MsgIssue.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgIssue();
        message.issuer = object.issuer ?? "";
        message.symbol = object.symbol ?? "";
        message.subunit = object.subunit ?? "";
        message.precision = object.precision ?? 0;
        message.initialAmount = object.initialAmount ?? "";
        message.description = object.description ?? "";
        message.features = object.features?.map((e) => e) || [];
        message.burnRate = object.burnRate ?? "";
        message.sendCommissionRate = object.sendCommissionRate ?? "";
        message.uri = object.uri ?? "";
        message.uriHash = object.uriHash ?? "";
        message.extensionSettings =
            object.extensionSettings !== undefined &&
                object.extensionSettings !== null
                ? exports.ExtensionIssueSettings.fromPartial(object.extensionSettings)
                : undefined;
        message.dexSettings =
            object.dexSettings !== undefined && object.dexSettings !== null
                ? token_1.DEXSettings.fromPartial(object.dexSettings)
                : undefined;
        return message;
    },
};
function createBaseExtensionIssueSettings() {
    return { codeId: 0, label: "", funds: [], issuanceMsg: new Uint8Array(0) };
}
exports.ExtensionIssueSettings = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.codeId !== 0) {
            writer.uint32(8).uint64(message.codeId);
        }
        if (message.label !== "") {
            writer.uint32(18).string(message.label);
        }
        for (const v of message.funds) {
            coin_1.Coin.encode(v, writer.uint32(26).fork()).ldelim();
        }
        if (message.issuanceMsg.length !== 0) {
            writer.uint32(34).bytes(message.issuanceMsg);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseExtensionIssueSettings();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 8) {
                        break;
                    }
                    message.codeId = longToNumber(reader.uint64());
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.label = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.funds.push(coin_1.Coin.decode(reader, reader.uint32()));
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.issuanceMsg = reader.bytes();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            codeId: isSet(object.codeId) ? globalThis.Number(object.codeId) : 0,
            label: isSet(object.label) ? globalThis.String(object.label) : "",
            funds: globalThis.Array.isArray(object?.funds)
                ? object.funds.map((e) => coin_1.Coin.fromJSON(e))
                : [],
            issuanceMsg: isSet(object.issuanceMsg)
                ? bytesFromBase64(object.issuanceMsg)
                : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.codeId !== 0) {
            obj.codeId = Math.round(message.codeId);
        }
        if (message.label !== "") {
            obj.label = message.label;
        }
        if (message.funds?.length) {
            obj.funds = message.funds.map((e) => coin_1.Coin.toJSON(e));
        }
        if (message.issuanceMsg.length !== 0) {
            obj.issuanceMsg = base64FromBytes(message.issuanceMsg);
        }
        return obj;
    },
    create(base) {
        return exports.ExtensionIssueSettings.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseExtensionIssueSettings();
        message.codeId = object.codeId ?? 0;
        message.label = object.label ?? "";
        message.funds = object.funds?.map((e) => coin_1.Coin.fromPartial(e)) || [];
        message.issuanceMsg = object.issuanceMsg ?? new Uint8Array(0);
        return message;
    },
};
function createBaseMsgMint() {
    return { sender: "", coin: undefined, recipient: "" };
}
exports.MsgMint = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.coin !== undefined) {
            coin_1.Coin.encode(message.coin, writer.uint32(18).fork()).ldelim();
        }
        if (message.recipient !== "") {
            writer.uint32(26).string(message.recipient);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgMint();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.coin = coin_1.Coin.decode(reader, reader.uint32());
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.recipient = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? globalThis.String(object.sender) : "",
            coin: isSet(object.coin) ? coin_1.Coin.fromJSON(object.coin) : undefined,
            recipient: isSet(object.recipient)
                ? globalThis.String(object.recipient)
                : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.coin !== undefined) {
            obj.coin = coin_1.Coin.toJSON(message.coin);
        }
        if (message.recipient !== "") {
            obj.recipient = message.recipient;
        }
        return obj;
    },
    create(base) {
        return exports.MsgMint.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgMint();
        message.sender = object.sender ?? "";
        message.coin =
            object.coin !== undefined && object.coin !== null
                ? coin_1.Coin.fromPartial(object.coin)
                : undefined;
        message.recipient = object.recipient ?? "";
        return message;
    },
};
function createBaseMsgBurn() {
    return { sender: "", coin: undefined };
}
exports.MsgBurn = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.coin !== undefined) {
            coin_1.Coin.encode(message.coin, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgBurn();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.coin = coin_1.Coin.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? globalThis.String(object.sender) : "",
            coin: isSet(object.coin) ? coin_1.Coin.fromJSON(object.coin) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.coin !== undefined) {
            obj.coin = coin_1.Coin.toJSON(message.coin);
        }
        return obj;
    },
    create(base) {
        return exports.MsgBurn.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgBurn();
        message.sender = object.sender ?? "";
        message.coin =
            object.coin !== undefined && object.coin !== null
                ? coin_1.Coin.fromPartial(object.coin)
                : undefined;
        return message;
    },
};
function createBaseMsgFreeze() {
    return { sender: "", account: "", coin: undefined };
}
exports.MsgFreeze = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.account !== "") {
            writer.uint32(18).string(message.account);
        }
        if (message.coin !== undefined) {
            coin_1.Coin.encode(message.coin, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgFreeze();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.account = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.coin = coin_1.Coin.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? globalThis.String(object.sender) : "",
            account: isSet(object.account) ? globalThis.String(object.account) : "",
            coin: isSet(object.coin) ? coin_1.Coin.fromJSON(object.coin) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.account !== "") {
            obj.account = message.account;
        }
        if (message.coin !== undefined) {
            obj.coin = coin_1.Coin.toJSON(message.coin);
        }
        return obj;
    },
    create(base) {
        return exports.MsgFreeze.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgFreeze();
        message.sender = object.sender ?? "";
        message.account = object.account ?? "";
        message.coin =
            object.coin !== undefined && object.coin !== null
                ? coin_1.Coin.fromPartial(object.coin)
                : undefined;
        return message;
    },
};
function createBaseMsgUnfreeze() {
    return { sender: "", account: "", coin: undefined };
}
exports.MsgUnfreeze = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.account !== "") {
            writer.uint32(18).string(message.account);
        }
        if (message.coin !== undefined) {
            coin_1.Coin.encode(message.coin, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUnfreeze();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.account = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.coin = coin_1.Coin.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? globalThis.String(object.sender) : "",
            account: isSet(object.account) ? globalThis.String(object.account) : "",
            coin: isSet(object.coin) ? coin_1.Coin.fromJSON(object.coin) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.account !== "") {
            obj.account = message.account;
        }
        if (message.coin !== undefined) {
            obj.coin = coin_1.Coin.toJSON(message.coin);
        }
        return obj;
    },
    create(base) {
        return exports.MsgUnfreeze.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgUnfreeze();
        message.sender = object.sender ?? "";
        message.account = object.account ?? "";
        message.coin =
            object.coin !== undefined && object.coin !== null
                ? coin_1.Coin.fromPartial(object.coin)
                : undefined;
        return message;
    },
};
function createBaseMsgSetFrozen() {
    return { sender: "", account: "", coin: undefined };
}
exports.MsgSetFrozen = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.account !== "") {
            writer.uint32(18).string(message.account);
        }
        if (message.coin !== undefined) {
            coin_1.Coin.encode(message.coin, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetFrozen();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.account = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.coin = coin_1.Coin.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? globalThis.String(object.sender) : "",
            account: isSet(object.account) ? globalThis.String(object.account) : "",
            coin: isSet(object.coin) ? coin_1.Coin.fromJSON(object.coin) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.account !== "") {
            obj.account = message.account;
        }
        if (message.coin !== undefined) {
            obj.coin = coin_1.Coin.toJSON(message.coin);
        }
        return obj;
    },
    create(base) {
        return exports.MsgSetFrozen.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgSetFrozen();
        message.sender = object.sender ?? "";
        message.account = object.account ?? "";
        message.coin =
            object.coin !== undefined && object.coin !== null
                ? coin_1.Coin.fromPartial(object.coin)
                : undefined;
        return message;
    },
};
function createBaseMsgGloballyFreeze() {
    return { sender: "", denom: "" };
}
exports.MsgGloballyFreeze = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.denom !== "") {
            writer.uint32(18).string(message.denom);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgGloballyFreeze();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.denom = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? globalThis.String(object.sender) : "",
            denom: isSet(object.denom) ? globalThis.String(object.denom) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.denom !== "") {
            obj.denom = message.denom;
        }
        return obj;
    },
    create(base) {
        return exports.MsgGloballyFreeze.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgGloballyFreeze();
        message.sender = object.sender ?? "";
        message.denom = object.denom ?? "";
        return message;
    },
};
function createBaseMsgGloballyUnfreeze() {
    return { sender: "", denom: "" };
}
exports.MsgGloballyUnfreeze = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.denom !== "") {
            writer.uint32(18).string(message.denom);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgGloballyUnfreeze();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.denom = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? globalThis.String(object.sender) : "",
            denom: isSet(object.denom) ? globalThis.String(object.denom) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.denom !== "") {
            obj.denom = message.denom;
        }
        return obj;
    },
    create(base) {
        return exports.MsgGloballyUnfreeze.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgGloballyUnfreeze();
        message.sender = object.sender ?? "";
        message.denom = object.denom ?? "";
        return message;
    },
};
function createBaseMsgClawback() {
    return { sender: "", account: "", coin: undefined };
}
exports.MsgClawback = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.account !== "") {
            writer.uint32(18).string(message.account);
        }
        if (message.coin !== undefined) {
            coin_1.Coin.encode(message.coin, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgClawback();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.account = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.coin = coin_1.Coin.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? globalThis.String(object.sender) : "",
            account: isSet(object.account) ? globalThis.String(object.account) : "",
            coin: isSet(object.coin) ? coin_1.Coin.fromJSON(object.coin) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.account !== "") {
            obj.account = message.account;
        }
        if (message.coin !== undefined) {
            obj.coin = coin_1.Coin.toJSON(message.coin);
        }
        return obj;
    },
    create(base) {
        return exports.MsgClawback.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgClawback();
        message.sender = object.sender ?? "";
        message.account = object.account ?? "";
        message.coin =
            object.coin !== undefined && object.coin !== null
                ? coin_1.Coin.fromPartial(object.coin)
                : undefined;
        return message;
    },
};
function createBaseMsgSetWhitelistedLimit() {
    return { sender: "", account: "", coin: undefined };
}
exports.MsgSetWhitelistedLimit = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.account !== "") {
            writer.uint32(18).string(message.account);
        }
        if (message.coin !== undefined) {
            coin_1.Coin.encode(message.coin, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetWhitelistedLimit();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.account = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.coin = coin_1.Coin.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? globalThis.String(object.sender) : "",
            account: isSet(object.account) ? globalThis.String(object.account) : "",
            coin: isSet(object.coin) ? coin_1.Coin.fromJSON(object.coin) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.account !== "") {
            obj.account = message.account;
        }
        if (message.coin !== undefined) {
            obj.coin = coin_1.Coin.toJSON(message.coin);
        }
        return obj;
    },
    create(base) {
        return exports.MsgSetWhitelistedLimit.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgSetWhitelistedLimit();
        message.sender = object.sender ?? "";
        message.account = object.account ?? "";
        message.coin =
            object.coin !== undefined && object.coin !== null
                ? coin_1.Coin.fromPartial(object.coin)
                : undefined;
        return message;
    },
};
function createBaseMsgTransferAdmin() {
    return { sender: "", account: "", denom: "" };
}
exports.MsgTransferAdmin = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.account !== "") {
            writer.uint32(18).string(message.account);
        }
        if (message.denom !== "") {
            writer.uint32(26).string(message.denom);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgTransferAdmin();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.account = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.denom = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? globalThis.String(object.sender) : "",
            account: isSet(object.account) ? globalThis.String(object.account) : "",
            denom: isSet(object.denom) ? globalThis.String(object.denom) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.account !== "") {
            obj.account = message.account;
        }
        if (message.denom !== "") {
            obj.denom = message.denom;
        }
        return obj;
    },
    create(base) {
        return exports.MsgTransferAdmin.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgTransferAdmin();
        message.sender = object.sender ?? "";
        message.account = object.account ?? "";
        message.denom = object.denom ?? "";
        return message;
    },
};
function createBaseMsgClearAdmin() {
    return { sender: "", denom: "" };
}
exports.MsgClearAdmin = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.denom !== "") {
            writer.uint32(18).string(message.denom);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgClearAdmin();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.denom = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? globalThis.String(object.sender) : "",
            denom: isSet(object.denom) ? globalThis.String(object.denom) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.denom !== "") {
            obj.denom = message.denom;
        }
        return obj;
    },
    create(base) {
        return exports.MsgClearAdmin.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgClearAdmin();
        message.sender = object.sender ?? "";
        message.denom = object.denom ?? "";
        return message;
    },
};
function createBaseMsgUpgradeTokenV1() {
    return { sender: "", denom: "", ibcEnabled: false };
}
exports.MsgUpgradeTokenV1 = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.denom !== "") {
            writer.uint32(18).string(message.denom);
        }
        if (message.ibcEnabled !== false) {
            writer.uint32(24).bool(message.ibcEnabled);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpgradeTokenV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.denom = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 24) {
                        break;
                    }
                    message.ibcEnabled = reader.bool();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? globalThis.String(object.sender) : "",
            denom: isSet(object.denom) ? globalThis.String(object.denom) : "",
            ibcEnabled: isSet(object.ibcEnabled)
                ? globalThis.Boolean(object.ibcEnabled)
                : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.denom !== "") {
            obj.denom = message.denom;
        }
        if (message.ibcEnabled !== false) {
            obj.ibcEnabled = message.ibcEnabled;
        }
        return obj;
    },
    create(base) {
        return exports.MsgUpgradeTokenV1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgUpgradeTokenV1();
        message.sender = object.sender ?? "";
        message.denom = object.denom ?? "";
        message.ibcEnabled = object.ibcEnabled ?? false;
        return message;
    },
};
function createBaseMsgUpdateParams() {
    return { authority: "", params: undefined };
}
exports.MsgUpdateParams = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.params !== undefined) {
            params_1.Params.encode(message.params, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.authority = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.params = params_1.Params.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            authority: isSet(object.authority)
                ? globalThis.String(object.authority)
                : "",
            params: isSet(object.params) ? params_1.Params.fromJSON(object.params) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.authority !== "") {
            obj.authority = message.authority;
        }
        if (message.params !== undefined) {
            obj.params = params_1.Params.toJSON(message.params);
        }
        return obj;
    },
    create(base) {
        return exports.MsgUpdateParams.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgUpdateParams();
        message.authority = object.authority ?? "";
        message.params =
            object.params !== undefined && object.params !== null
                ? params_1.Params.fromPartial(object.params)
                : undefined;
        return message;
    },
};
function createBaseMsgUpdateDEXUnifiedRefAmount() {
    return { sender: "", denom: "", unifiedRefAmount: "" };
}
exports.MsgUpdateDEXUnifiedRefAmount = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.denom !== "") {
            writer.uint32(18).string(message.denom);
        }
        if (message.unifiedRefAmount !== "") {
            writer.uint32(26).string(message.unifiedRefAmount);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateDEXUnifiedRefAmount();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.denom = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.unifiedRefAmount = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? globalThis.String(object.sender) : "",
            denom: isSet(object.denom) ? globalThis.String(object.denom) : "",
            unifiedRefAmount: isSet(object.unifiedRefAmount)
                ? globalThis.String(object.unifiedRefAmount)
                : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.denom !== "") {
            obj.denom = message.denom;
        }
        if (message.unifiedRefAmount !== "") {
            obj.unifiedRefAmount = message.unifiedRefAmount;
        }
        return obj;
    },
    create(base) {
        return exports.MsgUpdateDEXUnifiedRefAmount.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgUpdateDEXUnifiedRefAmount();
        message.sender = object.sender ?? "";
        message.denom = object.denom ?? "";
        message.unifiedRefAmount = object.unifiedRefAmount ?? "";
        return message;
    },
};
function createBaseMsgUpdateDEXWhitelistedDenoms() {
    return { sender: "", denom: "", whitelistedDenoms: [] };
}
exports.MsgUpdateDEXWhitelistedDenoms = {
    encode(message, writer = new binary_1.BinaryWriter()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.denom !== "") {
            writer.uint32(18).string(message.denom);
        }
        for (const v of message.whitelistedDenoms) {
            writer.uint32(26).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateDEXWhitelistedDenoms();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.sender = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.denom = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.whitelistedDenoms.push(reader.string());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? globalThis.String(object.sender) : "",
            denom: isSet(object.denom) ? globalThis.String(object.denom) : "",
            whitelistedDenoms: globalThis.Array.isArray(object?.whitelistedDenoms)
                ? object.whitelistedDenoms.map((e) => globalThis.String(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sender !== "") {
            obj.sender = message.sender;
        }
        if (message.denom !== "") {
            obj.denom = message.denom;
        }
        if (message.whitelistedDenoms?.length) {
            obj.whitelistedDenoms = message.whitelistedDenoms;
        }
        return obj;
    },
    create(base) {
        return exports.MsgUpdateDEXWhitelistedDenoms.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMsgUpdateDEXWhitelistedDenoms();
        message.sender = object.sender ?? "";
        message.denom = object.denom ?? "";
        message.whitelistedDenoms = object.whitelistedDenoms?.map((e) => e) || [];
        return message;
    },
};
function createBaseEmptyResponse() {
    return {};
}
exports.EmptyResponse = {
    encode(_, writer = new binary_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEmptyResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.EmptyResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseEmptyResponse();
        return message;
    },
};
exports.MsgServiceName = "coreum.asset.ft.v1.Msg";
class MsgClientImpl {
    constructor(rpc, opts) {
        this.service = opts?.service || exports.MsgServiceName;
        this.rpc = rpc;
        this.Issue = this.Issue.bind(this);
        this.Mint = this.Mint.bind(this);
        this.Burn = this.Burn.bind(this);
        this.Freeze = this.Freeze.bind(this);
        this.Unfreeze = this.Unfreeze.bind(this);
        this.SetFrozen = this.SetFrozen.bind(this);
        this.GloballyFreeze = this.GloballyFreeze.bind(this);
        this.GloballyUnfreeze = this.GloballyUnfreeze.bind(this);
        this.Clawback = this.Clawback.bind(this);
        this.SetWhitelistedLimit = this.SetWhitelistedLimit.bind(this);
        this.TransferAdmin = this.TransferAdmin.bind(this);
        this.ClearAdmin = this.ClearAdmin.bind(this);
        this.UpgradeTokenV1 = this.UpgradeTokenV1.bind(this);
        this.UpdateParams = this.UpdateParams.bind(this);
        this.UpdateDEXUnifiedRefAmount = this.UpdateDEXUnifiedRefAmount.bind(this);
        this.UpdateDEXWhitelistedDenoms =
            this.UpdateDEXWhitelistedDenoms.bind(this);
    }
    Issue(request) {
        const data = exports.MsgIssue.encode(request).finish();
        const promise = this.rpc.request(this.service, "Issue", data);
        return promise.then((data) => exports.EmptyResponse.decode(new binary_1.BinaryReader(data)));
    }
    Mint(request) {
        const data = exports.MsgMint.encode(request).finish();
        const promise = this.rpc.request(this.service, "Mint", data);
        return promise.then((data) => exports.EmptyResponse.decode(new binary_1.BinaryReader(data)));
    }
    Burn(request) {
        const data = exports.MsgBurn.encode(request).finish();
        const promise = this.rpc.request(this.service, "Burn", data);
        return promise.then((data) => exports.EmptyResponse.decode(new binary_1.BinaryReader(data)));
    }
    Freeze(request) {
        const data = exports.MsgFreeze.encode(request).finish();
        const promise = this.rpc.request(this.service, "Freeze", data);
        return promise.then((data) => exports.EmptyResponse.decode(new binary_1.BinaryReader(data)));
    }
    Unfreeze(request) {
        const data = exports.MsgUnfreeze.encode(request).finish();
        const promise = this.rpc.request(this.service, "Unfreeze", data);
        return promise.then((data) => exports.EmptyResponse.decode(new binary_1.BinaryReader(data)));
    }
    SetFrozen(request) {
        const data = exports.MsgSetFrozen.encode(request).finish();
        const promise = this.rpc.request(this.service, "SetFrozen", data);
        return promise.then((data) => exports.EmptyResponse.decode(new binary_1.BinaryReader(data)));
    }
    GloballyFreeze(request) {
        const data = exports.MsgGloballyFreeze.encode(request).finish();
        const promise = this.rpc.request(this.service, "GloballyFreeze", data);
        return promise.then((data) => exports.EmptyResponse.decode(new binary_1.BinaryReader(data)));
    }
    GloballyUnfreeze(request) {
        const data = exports.MsgGloballyUnfreeze.encode(request).finish();
        const promise = this.rpc.request(this.service, "GloballyUnfreeze", data);
        return promise.then((data) => exports.EmptyResponse.decode(new binary_1.BinaryReader(data)));
    }
    Clawback(request) {
        const data = exports.MsgClawback.encode(request).finish();
        const promise = this.rpc.request(this.service, "Clawback", data);
        return promise.then((data) => exports.EmptyResponse.decode(new binary_1.BinaryReader(data)));
    }
    SetWhitelistedLimit(request) {
        const data = exports.MsgSetWhitelistedLimit.encode(request).finish();
        const promise = this.rpc.request(this.service, "SetWhitelistedLimit", data);
        return promise.then((data) => exports.EmptyResponse.decode(new binary_1.BinaryReader(data)));
    }
    TransferAdmin(request) {
        const data = exports.MsgTransferAdmin.encode(request).finish();
        const promise = this.rpc.request(this.service, "TransferAdmin", data);
        return promise.then((data) => exports.EmptyResponse.decode(new binary_1.BinaryReader(data)));
    }
    ClearAdmin(request) {
        const data = exports.MsgClearAdmin.encode(request).finish();
        const promise = this.rpc.request(this.service, "ClearAdmin", data);
        return promise.then((data) => exports.EmptyResponse.decode(new binary_1.BinaryReader(data)));
    }
    UpgradeTokenV1(request) {
        const data = exports.MsgUpgradeTokenV1.encode(request).finish();
        const promise = this.rpc.request(this.service, "UpgradeTokenV1", data);
        return promise.then((data) => exports.EmptyResponse.decode(new binary_1.BinaryReader(data)));
    }
    UpdateParams(request) {
        const data = exports.MsgUpdateParams.encode(request).finish();
        const promise = this.rpc.request(this.service, "UpdateParams", data);
        return promise.then((data) => exports.EmptyResponse.decode(new binary_1.BinaryReader(data)));
    }
    UpdateDEXUnifiedRefAmount(request) {
        const data = exports.MsgUpdateDEXUnifiedRefAmount.encode(request).finish();
        const promise = this.rpc.request(this.service, "UpdateDEXUnifiedRefAmount", data);
        return promise.then((data) => exports.EmptyResponse.decode(new binary_1.BinaryReader(data)));
    }
    UpdateDEXWhitelistedDenoms(request) {
        const data = exports.MsgUpdateDEXWhitelistedDenoms.encode(request).finish();
        const promise = this.rpc.request(this.service, "UpdateDEXWhitelistedDenoms", data);
        return promise.then((data) => exports.EmptyResponse.decode(new binary_1.BinaryReader(data)));
    }
}
exports.MsgClientImpl = MsgClientImpl;
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
function longToNumber(int64) {
    const num = globalThis.Number(int64.toString());
    if (num > globalThis.Number.MAX_SAFE_INTEGER) {
        throw new globalThis.Error("Value is larger than Number.MAX_SAFE_INTEGER");
    }
    if (num < globalThis.Number.MIN_SAFE_INTEGER) {
        throw new globalThis.Error("Value is smaller than Number.MIN_SAFE_INTEGER");
    }
    return num;
}
function isSet(value) {
    return value !== null && value !== undefined;
}
