// coreum-js@2.17.4/dist/module/coreum/dex/v1/event.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.6.1
//   protoc               v3.21.12
// source: coreum-protos/dex/event.proto
/* eslint-disable */
import { BinaryReader, BinaryWriter } from "@bufbuild/protobuf/wire";
export const protobufPackage = "coreum.dex.v1";
function createBaseEventOrderPlaced() {
    return { creator: "", id: "", sequence: 0 };
}
export const EventOrderPlaced = {
    encode(message, writer = new BinaryWriter()) {
        if (message.creator !== "") {
            writer.uint32(10).string(message.creator);
        }
        if (message.id !== "") {
            writer.uint32(18).string(message.id);
        }
        if (message.sequence !== 0) {
            writer.uint32(24).uint64(message.sequence);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof BinaryReader ? input : new BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEventOrderPlaced();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.creator = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.id = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 24) {
                        break;
                    }
                    message.sequence = longToNumber(reader.uint64());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            creator: isSet(object.creator) ? globalThis.String(object.creator) : "",
            id: isSet(object.id) ? globalThis.String(object.id) : "",
            sequence: isSet(object.sequence) ? globalThis.Number(object.sequence) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.creator !== "") {
            obj.creator = message.creator;
        }
        if (message.id !== "") {
            obj.id = message.id;
        }
        if (message.sequence !== 0) {
            obj.sequence = Math.round(message.sequence);
        }
        return obj;
    },
    create(base) {
        return EventOrderPlaced.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEventOrderPlaced();
        message.creator = object.creator ?? "";
        message.id = object.id ?? "";
        message.sequence = object.sequence ?? 0;
        return message;
    },
};
function createBaseEventOrderReduced() {
    return { creator: "", id: "", sequence: 0, sentCoin: "", receivedCoin: "" };
}
export const EventOrderReduced = {
    encode(message, writer = new BinaryWriter()) {
        if (message.creator !== "") {
            writer.uint32(10).string(message.creator);
        }
        if (message.id !== "") {
            writer.uint32(18).string(message.id);
        }
        if (message.sequence !== 0) {
            writer.uint32(24).uint64(message.sequence);
        }
        if (message.sentCoin !== "") {
            writer.uint32(34).string(message.sentCoin);
        }
        if (message.receivedCoin !== "") {
            writer.uint32(42).string(message.receivedCoin);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof BinaryReader ? input : new BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEventOrderReduced();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.creator = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.id = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 24) {
                        break;
                    }
                    message.sequence = longToNumber(reader.uint64());
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.sentCoin = reader.string();
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.receivedCoin = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            creator: isSet(object.creator) ? globalThis.String(object.creator) : "",
            id: isSet(object.id) ? globalThis.String(object.id) : "",
            sequence: isSet(object.sequence) ? globalThis.Number(object.sequence) : 0,
            sentCoin: isSet(object.sentCoin)
                ? globalThis.String(object.sentCoin)
                : "",
            receivedCoin: isSet(object.receivedCoin)
                ? globalThis.String(object.receivedCoin)
                : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.creator !== "") {
            obj.creator = message.creator;
        }
        if (message.id !== "") {
            obj.id = message.id;
        }
        if (message.sequence !== 0) {
            obj.sequence = Math.round(message.sequence);
        }
        if (message.sentCoin !== "") {
            obj.sentCoin = message.sentCoin;
        }
        if (message.receivedCoin !== "") {
            obj.receivedCoin = message.receivedCoin;
        }
        return obj;
    },
    create(base) {
        return EventOrderReduced.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEventOrderReduced();
        message.creator = object.creator ?? "";
        message.id = object.id ?? "";
        message.sequence = object.sequence ?? 0;
        message.sentCoin = object.sentCoin ?? "";
        message.receivedCoin = object.receivedCoin ?? "";
        return message;
    },
};
function createBaseEventOrderCreated() {
    return {
        creator: "",
        id: "",
        sequence: 0,
        remainingBaseQuantity: "",
        remainingSpendableBalance: "",
    };
}
export const EventOrderCreated = {
    encode(message, writer = new BinaryWriter()) {
        if (message.creator !== "") {
            writer.uint32(10).string(message.creator);
        }
        if (message.id !== "") {
            writer.uint32(18).string(message.id);
        }
        if (message.sequence !== 0) {
            writer.uint32(24).uint64(message.sequence);
        }
        if (message.remainingBaseQuantity !== "") {
            writer.uint32(34).string(message.remainingBaseQuantity);
        }
        if (message.remainingSpendableBalance !== "") {
            writer.uint32(42).string(message.remainingSpendableBalance);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof BinaryReader ? input : new BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEventOrderCreated();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.creator = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.id = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 24) {
                        break;
                    }
                    message.sequence = longToNumber(reader.uint64());
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.remainingBaseQuantity = reader.string();
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.remainingSpendableBalance = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            creator: isSet(object.creator) ? globalThis.String(object.creator) : "",
            id: isSet(object.id) ? globalThis.String(object.id) : "",
            sequence: isSet(object.sequence) ? globalThis.Number(object.sequence) : 0,
            remainingBaseQuantity: isSet(object.remainingBaseQuantity)
                ? globalThis.String(object.remainingBaseQuantity)
                : "",
            remainingSpendableBalance: isSet(object.remainingSpendableBalance)
                ? globalThis.String(object.remainingSpendableBalance)
                : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.creator !== "") {
            obj.creator = message.creator;
        }
        if (message.id !== "") {
            obj.id = message.id;
        }
        if (message.sequence !== 0) {
            obj.sequence = Math.round(message.sequence);
        }
        if (message.remainingBaseQuantity !== "") {
            obj.remainingBaseQuantity = message.remainingBaseQuantity;
        }
        if (message.remainingSpendableBalance !== "") {
            obj.remainingSpendableBalance = message.remainingSpendableBalance;
        }
        return obj;
    },
    create(base) {
        return EventOrderCreated.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEventOrderCreated();
        message.creator = object.creator ?? "";
        message.id = object.id ?? "";
        message.sequence = object.sequence ?? 0;
        message.remainingBaseQuantity = object.remainingBaseQuantity ?? "";
        message.remainingSpendableBalance = object.remainingSpendableBalance ?? "";
        return message;
    },
};
function createBaseEventOrderClosed() {
    return {
        creator: "",
        id: "",
        sequence: 0,
        remainingBaseQuantity: "",
        remainingSpendableBalance: "",
    };
}
export const EventOrderClosed = {
    encode(message, writer = new BinaryWriter()) {
        if (message.creator !== "") {
            writer.uint32(10).string(message.creator);
        }
        if (message.id !== "") {
            writer.uint32(18).string(message.id);
        }
        if (message.sequence !== 0) {
            writer.uint32(24).uint64(message.sequence);
        }
        if (message.remainingBaseQuantity !== "") {
            writer.uint32(34).string(message.remainingBaseQuantity);
        }
        if (message.remainingSpendableBalance !== "") {
            writer.uint32(42).string(message.remainingSpendableBalance);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof BinaryReader ? input : new BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEventOrderClosed();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.creator = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.id = reader.string();
                    continue;
                }
                case 3: {
                    if (tag !== 24) {
                        break;
                    }
                    message.sequence = longToNumber(reader.uint64());
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.remainingBaseQuantity = reader.string();
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.remainingSpendableBalance = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            creator: isSet(object.creator) ? globalThis.String(object.creator) : "",
            id: isSet(object.id) ? globalThis.String(object.id) : "",
            sequence: isSet(object.sequence) ? globalThis.Number(object.sequence) : 0,
            remainingBaseQuantity: isSet(object.remainingBaseQuantity)
                ? globalThis.String(object.remainingBaseQuantity)
                : "",
            remainingSpendableBalance: isSet(object.remainingSpendableBalance)
                ? globalThis.String(object.remainingSpendableBalance)
                : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.creator !== "") {
            obj.creator = message.creator;
        }
        if (message.id !== "") {
            obj.id = message.id;
        }
        if (message.sequence !== 0) {
            obj.sequence = Math.round(message.sequence);
        }
        if (message.remainingBaseQuantity !== "") {
            obj.remainingBaseQuantity = message.remainingBaseQuantity;
        }
        if (message.remainingSpendableBalance !== "") {
            obj.remainingSpendableBalance = message.remainingSpendableBalance;
        }
        return obj;
    },
    create(base) {
        return EventOrderClosed.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEventOrderClosed();
        message.creator = object.creator ?? "";
        message.id = object.id ?? "";
        message.sequence = object.sequence ?? 0;
        message.remainingBaseQuantity = object.remainingBaseQuantity ?? "";
        message.remainingSpendableBalance = object.remainingSpendableBalance ?? "";
        return message;
    },
};
function longToNumber(int64) {
    const num = globalThis.Number(int64.toString());
    if (num > globalThis.Number.MAX_SAFE_INTEGER) {
        throw new globalThis.Error("Value is larger than Number.MAX_SAFE_INTEGER");
    }
    if (num < globalThis.Number.MIN_SAFE_INTEGER) {
        throw new globalThis.Error("Value is smaller than Number.MIN_SAFE_INTEGER");
    }
    return num;
}
function isSet(value) {
    return value !== null && value !== undefined;
}
