// coreum-js@2.17.4/dist/module/coreum/dex/v1/params.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.6.1
//   protoc               v3.21.12
// source: coreum-protos/dex/params.proto
/* eslint-disable */
import { BinaryReader, BinaryWriter } from "cosmjs-types/binary";
import { Coin } from "cosmjs-types/cosmos/base/v1beta1/coin";
export const protobufPackage = "coreum.dex.v1";
function createBaseParams() {
    return {
        defaultUnifiedRefAmount: "",
        priceTickExponent: 0,
        maxOrdersPerDenom: 0,
        orderReserve: undefined,
    };
}
export const Params = {
    encode(message, writer = new BinaryWriter()) {
        if (message.defaultUnifiedRefAmount !== "") {
            writer.uint32(10).string(message.defaultUnifiedRefAmount);
        }
        if (message.priceTickExponent !== 0) {
            writer.uint32(16).int32(message.priceTickExponent);
        }
        if (message.maxOrdersPerDenom !== 0) {
            writer.uint32(24).uint64(message.maxOrdersPerDenom);
        }
        if (message.orderReserve !== undefined) {
            Coin.encode(message.orderReserve, writer.uint32(34).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof BinaryReader ? input : new BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.defaultUnifiedRefAmount = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.priceTickExponent = reader.int32();
                    continue;
                }
                case 3: {
                    if (tag !== 24) {
                        break;
                    }
                    message.maxOrdersPerDenom = longToNumber(reader.uint64());
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.orderReserve = Coin.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            defaultUnifiedRefAmount: isSet(object.defaultUnifiedRefAmount)
                ? globalThis.String(object.defaultUnifiedRefAmount)
                : "",
            priceTickExponent: isSet(object.priceTickExponent)
                ? globalThis.Number(object.priceTickExponent)
                : 0,
            maxOrdersPerDenom: isSet(object.maxOrdersPerDenom)
                ? globalThis.Number(object.maxOrdersPerDenom)
                : 0,
            orderReserve: isSet(object.orderReserve)
                ? Coin.fromJSON(object.orderReserve)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.defaultUnifiedRefAmount !== "") {
            obj.defaultUnifiedRefAmount = message.defaultUnifiedRefAmount;
        }
        if (message.priceTickExponent !== 0) {
            obj.priceTickExponent = Math.round(message.priceTickExponent);
        }
        if (message.maxOrdersPerDenom !== 0) {
            obj.maxOrdersPerDenom = Math.round(message.maxOrdersPerDenom);
        }
        if (message.orderReserve !== undefined) {
            obj.orderReserve = Coin.toJSON(message.orderReserve);
        }
        return obj;
    },
    create(base) {
        return Params.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseParams();
        message.defaultUnifiedRefAmount = object.defaultUnifiedRefAmount ?? "";
        message.priceTickExponent = object.priceTickExponent ?? 0;
        message.maxOrdersPerDenom = object.maxOrdersPerDenom ?? 0;
        message.orderReserve =
            object.orderReserve !== undefined && object.orderReserve !== null
                ? Coin.fromPartial(object.orderReserve)
                : undefined;
        return message;
    },
};
function longToNumber(int64) {
    const num = globalThis.Number(int64.toString());
    if (num > globalThis.Number.MAX_SAFE_INTEGER) {
        throw new globalThis.Error("Value is larger than Number.MAX_SAFE_INTEGER");
    }
    if (num < globalThis.Number.MIN_SAFE_INTEGER) {
        throw new globalThis.Error("Value is smaller than Number.MIN_SAFE_INTEGER");
    }
    return num;
}
function isSet(value) {
    return value !== null && value !== undefined;
}
