// coreum-js@2.17.4/dist/module/coreum/customparams/v1/genesis.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.0
//   protoc               v3.21.12
// source: coreum-protos/customparams/genesis.proto
/* eslint-disable */
import { BinaryReader, BinaryWriter } from "cosmjs-types/binary";
import { StakingParams } from "./params";
export const protobufPackage = "coreum.customparams.v1";
function createBaseGenesisState() {
    return { stakingParams: undefined };
}
export const GenesisState = {
    encode(message, writer = new BinaryWriter()) {
        if (message.stakingParams !== undefined) {
            StakingParams.encode(message.stakingParams, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof BinaryReader ? input : new BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGenesisState();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.stakingParams = StakingParams.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            stakingParams: isSet(object.stakingParams)
                ? StakingParams.fromJSON(object.stakingParams)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.stakingParams !== undefined) {
            obj.stakingParams = StakingParams.toJSON(message.stakingParams);
        }
        return obj;
    },
    create(base) {
        return GenesisState.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGenesisState();
        message.stakingParams =
            object.stakingParams !== undefined && object.stakingParams !== null
                ? StakingParams.fromPartial(object.stakingParams)
                : undefined;
        return message;
    },
};
function isSet(value) {
    return value !== null && value !== undefined;
}
